agent/structs/config_entry_discoverychain.go

```diff
@@ -251,6 +257,26 @@ func isValidHTTPMethod(method string) bool {
 	}
 }
 
+func isValidRetryCondition(retryOn string) bool {
```

Comment:
It would be good to add a test for this validation function in agent/structs/config_entry_discoverychain_test.go to enumerate all the valid retry conditions. It is obvious that it works today but if we refactor things in the future it would be great to know if we break something.

Additionally it would be great if you could add two more cases to `TestServiceRouterConfigEntry` in that same file. One to ensure that a router with this field configured with correct entries is accepted and one to test the validation error.

Reply:
Updated. Let me know if the tests look fine now